metals/src/main/scala/scala/meta/internal/metals/FindTextInDependencyJars.scala

```diff
@@ -0,0 +1,110 @@
+package scala.meta.internal.metals
+
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.nio.file.Files
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.control.NonFatal
+
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.Location
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+
+class FindTextInDependencyJars(
+    buildTargets: BuildTargets,
+    workspace: () => AbsolutePath
+) {
+  def find(mask: String, content: String): List[Location] = {
+    val allLocations: ArrayBuffer[Location] = new ArrayBuffer[Location]()
+
+    buildTargets.allWorkspaceJars.foreach { classpathEntry =>
+      try {
+        val jarLocations: List[Location] =
+          if (classpathEntry.isFile && classpathEntry.isJar) {
+            visitJar(classpathEntry, mask, content)
+          } else Nil
+
+        allLocations ++= jarLocations
+      } catch {
+        case NonFatal(e) =>
+          scribe.error(
+            s"Failed to find text in dependency files for $classpathEntry",
+            e
+          )
+      }
+    }
+
+    allLocations.toList
+  }
+
+  private def isSuitableFile(path: AbsolutePath, mask: String): Boolean = {
+    path.isFile && path.filename.contains(mask)
```

Comment:
This should probably be a glob pattern instead of simply checking `contains`. 
```suggestion
    path.isFile && ScalafmtConfig.PathMatcher.Nio(mask).matches(path)
```

it coud make sense to move `ScalafmtConfig.PathMatcher` to just PathMatcher.

Reply:
With include/exclude we could do:
```scala
path.isFile && ScalafmtConfig.PathMatcher.Nio(include).matches(path) && !ScalafmtConfig.PathMatcher.Nio(exclude).matches(path)
```